File: app/src/main/java/org/kd/main/common/entities/Currency.java
package org.kd.main.common.entities;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@NoArgsConstructor
@Getter
@Setter
@AllArgsConstructor
public class Currency implements Serializable {

    @Id
    private Long id;

    private String name;
    private String shortname;

    public Currency(String name, String shortname) {
        this.name = name;
        this.shortname = shortname;
    }

    @Override
    public String toString() {
        return shortname;
    }
}

File: C#UI/Banque/Common/Entities/Currency.cs
using System;

namespace Banque.Common.Entities
{
    public class Currency
    {
        public long Id { get; set; }
        public String Name { get; set; }
        public String Shortname { get; set; }

        public Currency()
        {
        }

        public Currency(String name, String shortname)
        {
            this.Name = name;
            this.Shortname = shortname;
        }

        public Currency(long id, String name, String shortname)
        {
            this.Id = id;
            this.Name = name;
            this.Shortname = shortname;
        }

        override
        public String ToString()
        {
            return Shortname;
        }
    }
}
